File: client/src/app/model/prato.ts
import { Restaurante } from "./restaurante";

export class Prato {
  id: number;
  restaurante: Restaurante;
  descricao: string;
  valor: number;

  constructor(params?: Partial<Prato>) {
    if (params) {
      this.id = params.id;
      this.restaurante = params.restaurante;
      this.descricao = params.descricao;
      this.valor = params.valor;
    }
  }
}

File: web-service/src/main/java/br/com/lucas/simpledinner/model/Prato.java
package br.com.lucas.simpledinner.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "prato")
public class Prato {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne(targetEntity = Restaurante.class)
    @JoinColumn(name="id_restaurante")
    private Restaurante restaurante;

    @Column(name="descricao")
    private String descricao;

    @Column(name="valor")
    private Double valor;

    public Prato() {
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public Restaurante getRestaurante() {
        return restaurante;
    }

    public void setRestaurante(Restaurante restaurante) {
        this.restaurante = restaurante;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public Double getValor() {
        return valor;
    }

    public void setValor(Double valor) {
        this.valor = valor;
    }

}
